### PR TITLE
Update pre-commit, py 3.10 support and support for vmotion work-in-progress

### PR DESCRIPTION
**Pull request by dmurphy18**
Updated the pre-commit for updates similar to Salt
Added support for Python 3.10 to pre-commit and testing
Added support for vmotion get|enable|disable - work-in-progress, using it to test the pre-commit and py3.10 changes

### CODE REVIEW

**Comment by dmurphy18** [comment]
Note: disabled Sphinx coverage use since 7.2.6 has a divide by zero issue, which had been fixed and should be available when Sphinx v7.3.0 is released

File: setup.cfg [hunk 1]
```diff
@@ -47,7 +48,7 @@ install_requires =
 tests =
   pytest>=6.1.0
   pytest-cov
-  pytest-salt-factories>=0.121.1
+  pytest-salt-factories>=0.912.2
```
Comment (s0undt3ch) [inline setup.cfg:51]: It might require quite a few updates but these days the version that should be getting used is the 1.0.0rcN version

Comment (dmurphy18) [reply setup.cfg:51]: Would update pytest-salt-factories to 1.0.0rcN, but the version available from PyPI is 0.912.2

Comment (dmurphy18) [reply setup.cfg]: updated to 1.0.0rc27

Comment (s0undt3ch) [reply setup.cfg]: You might have other stuff to fix on tests now

**Review by garethgreenaway — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**